src/media-segment-request.js

```diff
@@ -0,0 +1,281 @@
+import videojs from 'video.js';
+import { createTransferableMessage } from './bin-utils';
+
+export const REQUEST_ERRORS = {
+  ABORTED: -101,
+  TIMEOUT: -102,
+  FAILURE: 2
+};
+
+/**
+ * Turns segment byterange into a string suitable for use in
+ * HTTP Range requests
+ */
+const byterangeStr = function(byterange) {
+  let byterangeStart;
+  let byterangeEnd;
+
+  // `byterangeEnd` is one less than `offset + length` because the HTTP range
+  // header uses inclusive ranges
+  byterangeEnd = byterange.offset + byterange.length - 1;
+  byterangeStart = byterange.offset;
+  return 'bytes=' + byterangeStart + '-' + byterangeEnd;
+};
+
+/**
+ * Defines headers for use in the xhr request for a particular segment.
+ */
+const segmentXhrHeaders = function(segment) {
+  let headers = {};
+
+  if (segment.byterange) {
+    headers.Range = byterangeStr(segment.byterange);
+  }
+  return headers;
+};
+
+const abortAll = (activeXhrs) => {
+  Object.keys(activeXhrs).forEach((xhrKey) => {
+    const xhr = activeXhrs[xhrKey];
+
+    // only abort xhrs that haven't had a response
+    if (!xhr.responseTime) {
```

Comment:
Does it hurt to call `abort` on a request that has already been aborted? If so should probably add a check for that here as well. If one of the requests causes an error, the rest of the requests are aborted which in turn would cause an error on the aborted request, which would loop through all the requests again, even the aborted one